crates/llms/src/embeddings/candle/tei.rs

```diff
@@ -127,20 +129,36 @@ impl TeiEmbed {
             })
             .transpose()?
             .flatten();
-        let model_root = download_hf_artifacts(model_id, revision, hf_token)?;
-        Self::from_dir(&model_root, pool)
+        let model_root = download_hf_artifacts(model_id, revision, hf_token).await?;
+        Self::from_dir(&model_root, pool, max_seq_length_overwrite)
     }
 
     /// Instantiates a text-embedding-inference service with model, tokenizer, config, etc files in a single directory.
-    pub fn from_dir(root: &Path, pooling_overwrite: Option<Pool>) -> Result<Self> {
+    pub fn from_dir(
+        root: &Path,
+        pooling_overwrite: Option<Pool>,
+        max_seq_length_overwrite: Option<usize>,
+    ) -> Result<Self> {
         let tokenizer = load_tokenizer(root)?;
         let config = load_config(root)?;
 
         // Load [`Tokenization`]
         let position_offset = position_offset(&config);
-        // TODO: `max_input_length` should take into account overwrite files like `sentence_bert_config.json`.
-        // See `<https://github.com/huggingface/text-embeddings-inference/blob/cb1e594709fb1caea674ed460b6e426b2b4a531b/router/src/lib.rs#L189>`
-        let max_input_length = config.max_position_embeddings - position_offset;
+
+        let max_input_length = if let Some(max_seq_length) = max_seq_length_overwrite {
+            max_seq_length
+        } else {
+            // Some models will have `sentence_*_config.json` file defining a specific `max_seq_length`.
+            match max_seq_length_from_st_config(root) {
+                Ok(Some(max_seq_length)) => max_seq_length,
```

Comment:
The following version is easier to read or understand IMO (assume `_ =>` refers to Ok(None)):
Ok(max_seq_length) => max_seq_length.unwrap_or(config.max_position_embeddings - position_offset)